docs/index.md

```diff
@@ -1,4 +1,49 @@
 # Architecture
 
-Skipper is written as library and has also 2 binaries, named `skipper`
-and `eskip`.
+Skipper is written as library and is also a multi binary project with
+2 binaries, named `skipper` and `eskip`. `Skipper` is the HTTP proxy
+and `eskip` is a CLI application to verify, print, update or delete
+Skipper routes.
+
+Skipper's internal architecture is split into different packages. The
+`skipper` package has connections to multiple `dataclient`, that pull
+information from different sources, for example static routes from an
+eskip file or dynamic routes from Kubernetes ingress objects.
+
+The `proxy` package gets the routes populated by skipper and has
+always a current routing table which will be replaced on change.
+
+A route is one entry in the routing table. A route consists of one or
+more `predicate`, that are used to find a route for a given HTTP
+request. A route can also have one or more `filter`, that can modify
+the content of the request or response.
```

Comment:
I'd add here something like: A route always points to a backend, or it's a shunt, meaning that skipper serves the requests for the route, or a loopback, meaning that the request will be matched against the routing table again after filters have modified it.

Reply:
I will use `<shunt>` and the same for loopback